### PR TITLE
aws-lambda: Add function url types

### PR DESCRIPTION
**Pull request by milesnash**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.aws.amazon.com/lambda/latest/dg/urls-invocation.html

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/aws-lambda/trigger/lambda-function-url.d.ts
1. types/aws-lambda/test/lambda-function-url-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 26df14331e66972b78f16c42e9f6fb35f0711d63
</p>

**Comment by simonbuchan** [comment]
Hmm, I appreciate making it clear that lambda urls use the same format as API gateway http, but this seems a bit much?I think simply adding the aliases to the API gateway trigger and the test could even delegate to the API gateway test...

I'll defer to other reviewers for an actual approval or not though.

File: types/aws-lambda/trigger/lambda-function-url.d.ts
Comment (joeykilpatrick) [inline types/aws-lambda/trigger/lambda-function-url.d.ts:3]: Can we get Event, Result, and Callback aliases added as well? I think many people use those types directly, not just the Handler types.

**Review by joeykilpatrick — CHANGES_REQUESTED** [review]
I think it's fair to say that these types probably won't diverge from the the API Gateway Proxy types. Future changes to those types shouldn't require changes in these test files as well. I would be in favor of removing the tests but leaving this as an alias.

File: types/aws-lambda/trigger/lambda-function-url.d.ts
Comment (milesnash) [reply types/aws-lambda/trigger/lambda-function-url.d.ts:3]: > Can we add Event, Result, and Callback aliases added as well? I think many people use those types directly, not just the Handler types.

Hi @joeykilpatrick , thanks for your feedback, I have removed the duplicated tests as suggested in your other comment.  On this point, could I clarify if you also mean that we should implement `LambdaFunctionURLHandler` separately (i.e. not as an alias of `APIGatewayProxyHandlerV2`) as a result of this suggestion?  Because if we are to alias the Event, Result, and Callback interfaces, then I think it would be weird to _not_ do that since would mean `LambdaFunctionURLHandler` is still essentially `Handler<APIGatewayProxyEventV2, APIGatewayProxyResultV2<T>>`, which would be inconsistent with those interfaces.

Just to be clear, I think it's a good suggestion, but I think it will ultimately involve more interfaces/types that those three if you inspect the interface/type hierarchy of the API Gateway types, unless there's some TS wizardry that I'm overlooking 🙂

Comment (joeykilpatrick) [reply types/aws-lambda/trigger/lambda-function-url.d.ts:3]: This is what I mean:
```typescript
export type LambdaFunctionURLEvent = APIGatewayProxyEventV2;
export type LambdaFunctionURLResult<T = never> = APIGatewayProxyResultV2<T>;
export type LambdaFunctionURLCallback = APIGatewayProxyCallbackV2;
export type LambdaFunctionURLHandler<T = never> = APIGatewayProxyHandlerV2<T>
```

Comment (milesnash) [reply types/aws-lambda/trigger/lambda-function-url.d.ts:28]: > This is what I mean:
> 
> ```ts
> export type LambdaFunctionURLEvent = APIGatewayProxyEventV2;
> export type LambdaFunctionURLResult<T = never> = APIGatewayProxyResultV2<T>;
> export type LambdaFunctionURLCallback = APIGatewayProxyCallbackV2;
> export type LambdaFunctionURLHandler<T = never> = APIGatewayProxyHandlerV2<T>
> ```

Ah ok, I wondered about the consistency of the types below the handlers in that case but I'm struggling to communicate that effectively.  Hopefully my most recent commit gets across what I'm trying to explain here.

Comment (joeykilpatrick) [reply types/aws-lambda/trigger/lambda-function-url.d.ts:16]: Changes look good. Once the pipeline is fixed, I will approve.

**Review by joeykilpatrick — APPROVED** [review]

File: types/aws-lambda/trigger/lambda-function-url.d.ts
Comment (milesnash) [reply types/aws-lambda/trigger/lambda-function-url.d.ts:16]: Great, thanks for your feedback! 🚀

**Comment by joeykilpatrick** [comment]
Ping, just to keep this PR open. Waiting for review by DT maintainer.

**Review by gabritto — APPROVED** [review]

**Comment by joeykilpatrick** [comment]
Ready to merge

**Pull request merged**
